### PR TITLE
memory and task code cleanup

### PR DESCRIPTION
**Pull request by xiaoxmeng**
Remove CACHE and TEST kinds and use the kind of the delegated
memory allocator instead. In debugging, it is more meaningful log
out the actual kind of the delegated memory allocator.

Rename STD to MALLOC and will do the class name cleanup in followup.

### CODE REVIEW

File: velox/common/memory/MemoryAllocator.cpp [hunk 1]
```diff
@@ -381,6 +387,12 @@ bool MemoryAllocatorImpl::allocateContiguousImpl(
     } catch (std::exception& e) {
       // If the new memory reservation fails, we need to release the memory
       // reservation of the freed contiguous and non-contiguous memory.
+      LOG(ERROR) << "Failed to reserve " << numNeededPages * kPageSize
```
Comment (mbasmanova) [inline velox/common/memory/MemoryAllocator.cpp:390]: Should this be a warning and not error. I assume we expect this to happen in tight memory environments.

File: velox/common/memory/MemoryAllocator.h [hunk 2]
```diff
@@ -164,23 +164,12 @@ class MemoryAllocator : public std::enable_shared_from_this<MemoryAllocator> {
...
     /// manages the large chunk of memory allocations on its own by leveraging
     /// mmap and madvice, to optimize the memory fragmentation in the long
     /// running service such as Prestissimo.
     kMmap,
-    /// The memory allocator kind which is implemented by AsyncDataCache to
-    /// integrate with the file cache management. It is a wrapper on top of an
-    /// actual memory allocator. It delegates the memory allocation to the
-    /// associated memory allocator and might evict file cache to make space for
-    /// the memory allocations and retries on allocation failures.
-    kCache,
```
Comment (mbasmanova) [inline velox/common/memory/MemoryAllocator.h:178]: What's the motivation for this change? Would you update PR description to clarify?

File: velox/common/memory/MemoryAllocator.h [hunk 3]
```diff
@@ -261,9 +250,7 @@ class MemoryAllocator : public std::enable_shared_from_this<MemoryAllocator> {
       pool_ = other.pool_;
       runs_ = std::move(other.runs_);
       numPages_ = other.numPages_;
-      other.numPages_ = 0;
-      other.runs_.clear();
-      other.pool_ = nullptr;
+      other.clear();
```
Comment (mbasmanova) [inline velox/common/memory/MemoryAllocator.h:253]: Nice refactor.

File: velox/common/memory/MemoryAllocator.h [hunk 4]
```diff
@@ -378,12 +360,13 @@ class MemoryAllocator : public std::enable_shared_from_this<MemoryAllocator> {
       return size_;
     }
 
-    void setPool(MemoryPool* FOLLY_NONNULL pool) {
+    void setPool(MemoryPool* pool) {
```
Comment (mbasmanova) [inline velox/common/memory/MemoryAllocator.h:363]: I wonder if it makes sense to remove this setter and pass the pool to the constructor. Would you document this method to explain how it is supposed to be used? Is it Ok to never set the pool?

File: velox/exec/Task.cpp [hunk 5]
```diff
@@ -792,34 +792,33 @@ std::unique_ptr<ContinuePromise> Task::addSplitLocked(
...
-      seenSplitGroups_.emplace(splitGroupId);
-      queuedSplitGroups_.push(splitGroupId);
-      auto self = shared_from_this();
-      // We might have some free driver slots to process this split group.
-      ensureSplitGroupsAreBeingProcessedLocked(self);
-    }
-    return addSplitToStoreLocked(
-        splitsState.groupSplitsStores[splitGroupId], std::move(split));
   }
+  VELOX_CHECK(split.hasGroup(), "Missing split group for grouped execution!");
```
Comment (mbasmanova) [inline velox/exec/Task.cpp:796]: nit: add empty line before this statement

Should this be a USER_CHECK? Perhaps, add task ID to the error message.

**Review by mbasmanova — APPROVED** [review]
@xiaoxmeng Looks good % some questions below.

File: velox/common/memory/MemoryAllocator.h (hunk 2 above)
Comment (xiaoxmeng) [reply velox/common/memory/MemoryAllocator.h:178]: I put it PR description. Thanks!

File: velox/common/memory/MemoryAllocator.h (hunk 4 above)
Comment (xiaoxmeng) [reply velox/common/memory/MemoryAllocator.h:363]: We can't set it in constructor as it is also used in memory allocator internally for temporal memory allocations which we don't have the owner pool. I previously thought the owner could either be a allocator or pool but it is a little bit complicated. The memory pool is the allocation owner of all external use cases and all the allocations go through memory pool and we set the owner there on success. I will put the description there.
